pandas/tests/arrays/test_boolean.py

```diff
@@ -391,13 +391,101 @@ def test_scalar(self, data, all_logical_operators):
 
     def test_array(self, data, all_logical_operators):
         op_name = all_logical_operators
+        if "or" in op_name:
+            pytest.skip("confusing")
         other = pd.array([True] * len(data), dtype="boolean")
         self._compare_other(data, op_name, other)
         other = np.array([True] * len(data))
         self._compare_other(data, op_name, other)
         other = pd.Series([True] * len(data), dtype="boolean")
         self._compare_other(data, op_name, other)
 
+    def test_kleene_or(self):
```

Comment:
A careful review of these new test cases would be greatly appreciated. I've tried to make them as clear as possible, while covering all the cases.

Reply:
I went through the tests, very clear, added a few comments, for the rest looks good to me!